### PR TITLE
[Dynamo] graph break on isinstance calls if we don't know the type

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #118928
* __->__ #118778
* #118768

If we can't figure out the python type of a VariableTracker, then the
isinstance call should graph break (instead of raising an error).

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Comment by zou3519** [comment]
Test failures are me needing to update dynamo_test_failures.py (RIP)

**Review by ydwu4 — APPROVED** [review]

File: torch/_dynamo/variables/builtin.py [hunk 1]
```diff
@@ -1053,7 +1053,12 @@ def call_getitem(self, tx, *args, **kwargs):
         return args[0].call_method(tx, "__getitem__", args[1:], kwargs)
 
     def call_isinstance(self, tx, arg, isinstance_type):
-        arg_type = arg.python_type()
+        try:
+            arg_type = arg.python_type()
+        except NotImplementedError:
+            unimplemented(
+                "isinstance({arg}, {isinstance_type}): can't determine type of {arg}"
```
Comment (ydwu4) [inline torch/_dynamo/variables/builtin.py:1060]: nit: missing a `f` prefix?

Comment (zou3519) [reply torch/_dynamo/variables/builtin.py:1060]: thanks, good catch

**Pull request closed without merging**